libcst/metadata/scope_provider.py

```diff
@@ -789,10 +789,8 @@ def __init__(self, provider: "ScopeProvider") -> None:
         self.scope: Scope = GlobalScope()
         self.__deferred_accesses: List[DeferredAccess] = []
         self.__top_level_attribute_stack: List[Optional[cst.Attribute]] = [None]
-        self.__in_annotation: Set[
-            Union[cst.Call, cst.Annotation, cst.Subscript]
-        ] = set()
-        self.__in_type_hint: Set[Union[cst.Call, cst.Annotation, cst.Subscript]] = set()
+        self.__in_annotation_stack: List[bool] = [False]
+        self.__in_type_hint_stack: List[bool] = [False]
         self.__in_ignored_subscript: Set[cst.Subscript] = set()
```

Comment:
I actually think I can remove `__in_ignored_subscript` and implement it by pushing False into `__in_annotation_stack`